### PR TITLE
docs(examples) Add table and improve design

### PR DESCRIPTION
**Pull request by charlesbvll**
### Description

Currently the Flower Examples doc page is just an organized list of pages, without any description what so ever:

image

### Related issues/PRs

N/A

### Explanation

- Use Python instead of Bash
- Add intro/description of what Flower Examples are
- Use a table to display the list of pages
- Separate the examples into `quickstart`, `advanced`, and `others`

image

### Checklist

- [x] Implement proposed change
- [x] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [x] Make CI checks pass
- [x] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### Any other comments?

N/A

### CODE REVIEW

File: examples/app-secure-aggregation/README.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+---
+title: Example Flower App with Secure Aggregation
+labels: [basic, vision, fds]
+dataset: [CIFAR-10 | https://huggingface.co/datasets/uoft-cs/cifar10]
```
Comment (jafermarq) [inline examples/app-secure-aggregation/README.md:4]: This example doesn't use a dataset (just `NumPy` array simulating a model being sent). @panh99 what should we do with this example?

File: examples/embedded-devices/README.md [hunk 2]
```diff
@@ -1,3 +1,10 @@
+---
+title: Flower Embedded Devices Example
+labels: [basic, vision, fds]
+dataset: [CIFAR-10 | https://huggingface.co/datasets/uoft-cs/cifar10]
```
Comment (jafermarq) [inline examples/embedded-devices/README.md:4]: This example can be run with both `CIFAR-10` and `MNIST`.

File: examples/fl-dp-sa/README.md [hunk 3]
```diff
@@ -1,4 +1,11 @@
-# fl_dp_sa
+---
+title: Example of Flower App with DP and SA
+labels: [basic, vision, fds]
+dataset: [CIFAR-10 | https://huggingface.co/datasets/uoft-cs/cifar10]
```
Comment (jafermarq) [inline examples/fl-dp-sa/README.md:4]:
```suggestion
dataset: [MNIST | https://huggingface.co/datasets/ylecun/mnist]
```

File: examples/fl-tabular/README.md [hunk 4]
```diff
@@ -1,3 +1,10 @@
+---
+title: Flower Example on Adult Census Income Tabular Dataset
+labels: [basic, tabular, fds]
+dataset: [Adult Census Income | https://www.kaggle.com/datasets/uciml/adult-census-income/data]
+framework: [scikit-learn | https://scikit-learn.org/]
```
Comment (jafermarq) [inline examples/fl-tabular/README.md:5]:
```suggestion
framework: [scikit-learn | https://scikit-learn.org/, torch | https://pytorch.org/]
```

File: examples/flower-via-docker-compose/README.md [hunk 5]
```diff
@@ -1,3 +1,10 @@
+---
+title: Leveraging Flower and Docker for Device Heterogeneity Management in FL
+labels: [deployment, vision, tutorial]
+dataset: [CIFAR-10 | https://huggingface.co/datasets/uoft-cs/cifar10]
+framework: [Docker | https://www.docker.com/]
```
Comment (jafermarq) [inline examples/flower-via-docker-compose/README.md:5]:
```suggestion
framework: [Docker | https://www.docker.com/, tensorflow | https://www.tensorflow.org/]
```

File: examples/llm-flowertune/README.md [hunk 6]
```diff
@@ -1,3 +1,10 @@
+---
+title: Federated LLM Fine-tuning with Flower
+labels: [llm, nlp, LLama2]
+dataset: [Alpaca-GPT4 | https://huggingface.co/datasets/vicgalle/alpaca-gpt4]
+framework: [PEFT | https://huggingface.co/docs/peft/index]
```
Comment (jafermarq) [inline examples/llm-flowertune/README.md:5]:
```suggestion
framework: [PEFT | https://huggingface.co/docs/peft/index, torch | https://pytorch.org/]
```

File: examples/embedded-devices/README.md (hunk 2 above)
Comment (jafermarq) [inline examples/embedded-devices/README.md:4]:
```suggestion
dataset: [CIFAR-10 | https://huggingface.co/datasets/uoft-cs/cifar10, MNIST | https://huggingface.co/datasets/ylecun/mnist]
```

Comment (jafermarq) [reply examples/embedded-devices/README.md:4]: it can be run with both.

File: examples/app-secure-aggregation/README.md (hunk 1 above)
Comment (charlesbvll) [reply examples/app-secure-aggregation/README.md]: I made it so it is possible to leave the dataset empty

File: dev/build-example-docs.py [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+.. button-link:: https://flower.ai/join-slack
+    :color: primary
+    :shadow:
+
+    Join us on Slack
+
+Quickstart Examples
+-------------------
+
+Flower Quickstart Examples are a collection of demo project that show how you
```
Comment (jafermarq) [inline dev/build-example-docs.py:48]:
```suggestion
Flower Quickstart Examples are a collection of demo projects that show how you
```

File: dev/build-example-docs.py [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+            "Federated Learning but also somewhat familiar with Flower's main "
+            "features.\n"
+        )
+        index_file.write(categories["advanced"]["table"])
+
+        index_file.write("\nOther Examples\n--------------\n")
+        index_file.write(
+            "Flower Examples are a collection of example projects written with "
+            "Flower that explore different domains and features. You can check "
+            "which examples already exist and/or contribute your own example.\n"
```
Comment (jafermarq) [inline dev/build-example-docs.py:210]: should we add a link to the "contribute your own example" text pointing to our contributors guide?

**Review by danieljanes — APPROVED** [review]

**Pull request merged**